### ISSUE TITLE
.NetStandard2.0 version

### ISSUE THREAD

**Issue by sescandell** [autofac/Autofac#889]
Hi there,

Any change to get a release including .netstandard2.0 target ?

If I well understand latests commits and issues, things are made so .netstandard2.0 is now targeted.

I made a [simple test project](https://github.com/sescandell/autofac-azurefunctionv2-error) and for now, we cannot use Azure Function V2 with Autofac, I supposed because of .netstandard2 targeting issue (at runtime, process doesn't succeed to load the Autofac Assembly).

Thanks for your feedback,

**Comment by tillig** [comment]
Duplicate of #867.

netstandard 1.1 is 100% compatible with netstandard 2.0. The reverse is not true - you can't use netstandard 2.0 items in netstandard 1.1 projects. We need to maintain as wide a compatibility as possible, so we will continue targeting netstandard 1.1. If that isn't working in your netstandard 2.0 project, it's likely a tooling problem. I'd recommend asking the folks in the [Azure Functions repo](https://github.com/Azure/Azure-Functions) or on StackOverflow.

Note [a quick Google search of "azure functions autofac"](https://www.google.com/search?q=azure+functions+autofac) returns several things that may help you.

- [Here's a blog article with a source link that shows a person who got this working](https://devkimchi.com/2017/11/16/azure-functions-with-ioc-container/)
- [There's a whole demo repo someone put up showing Autofac working with Azure functions](https://github.com/introtocomputerscience/azure-function-autofac-dependency-injection)
- [Here's an Azure Functions issue where they specifically say there's not really any DI support in Azure Functions](https://github.com/Azure/Azure-Functions/issues/299)

I'd recommend doing some searching to see if you can get unblocked before filing an issue with Azure Functions or asking on StackOverflow. It looks like the examples and answers you're looking for could be out there already.

**Comment by MihaMarkic** [comment]
@tillig While this is true, there is a drawback not having .NETStandard 2.0 target: it will drag a lot of references to a (i.e. Xamarin) project. Supporting .NETStandard 2.0 means just adding a target to <TargetFrameworks>netstandard2.0</> and fixing a empty array allocation. The code will be still the same but it won't drag a ton of references.
Also @sescandell asked to *add* target, not to replace it.

I can provide a PR if you wish so.

**Comment by alexmg** [comment]
The empty array return value could actually be `Enumerable.Empty<IComponentRegistration>()` which is the normal return value used when no match is found in a `RegistrationsFor` method. It explains why only that one compiler issue was found.

The general recommendation is "to target the lowest version of .NET Standard possible".

@MihaMarkic Can you expand on the issue you are seeing with the Xamarin project?

@sescandell Did you manage to find out what the cause of your Azure Functions V2 issue was?

If there are good reasons for adding an additional target then it is something we can consider. If the issue lies elsewhere then the problem will only be solved until another library targeting a lower version of .NET Standard is added to such projects triggering the same issue.

**Comment by MihaMarkic** [comment]
@alexmg Check [references](https://misc.rthand.com/autofac_xamarin.png).

**Comment by sescandell** [comment]
Hi,

I [opened an issue](https://social.msdn.microsoft.com/Forums/azure/en-US/d5470cd1-2df3-4907-aa09-7028e484b910/afv2-load-netstandard11-assembly-failed?forum=AzureFunctions) on MSDN Forum.

This specific issue seems to be tied to Azure Function V2 in particular. As I said in my first post, everything is working fine on Azure Function V1. I'm not asking "how to make Autofac work with Azure Function". Something that worked with AFv1 doesn't work anymore with AFv2.

But, contrary to my first feelings and as @tillig precised, this is not an issue from Autofac neither because of a missing targeted framework. Azure Function v2 is working fine with Autofac v4.0.0. 
Microsoft probably changed something on how libraries are loaded that break using a different version of Autofac than the one embedded by Azure Functions.

Thanks for your help!

**Comment by MihaMarkic** [comment]
By the way, I made a few tiny modifications to sources for targeting :NET Standard 2.0. Seems working just fine in a project of mine.

**Comment by MihaMarkic** [comment]
Just to end the debate. Latest VS NuGet pacakage manager will hide the mentioned references and thus won't pollute anymore.
IOW no need for support .NET Standard 2 anymore from this point of view.
